overlord/snapstate/backend/link.go

```diff
@@ -405,3 +410,16 @@ func (b Backend) UnlinkComponent(cpi snap.ContainerPlaceInfo, snapRev snap.Revis
 
 	return nil
 }
+
+func (b Backend) KillSnapApps(snapName string, reason snap.AppKillReason, meter progress.Meter) error {
```

Comment:
would it make sense to track timings here?

Reply:
yes, similar to starting/stopping service, updated!

Reply:
Also thanks for pointing this out, as I found a nasty bug while adding timings where basically the kill task gets stuck due to locked state because I was using `NewTaskProgressAdapterUnlocked`.

I found that there was no need to lock the state for killing apps or stopping services so I unlocked the state altogether like what is being done `stop-snap-services`. https://github.com/snapcore/snapd/pull/14189/commits/ba67be0451a4ad10f3680d2edd9ae4df206b95db